core/actions/index.ts

```diff
@@ -96,15 +97,52 @@ export abstract class ActionBuilder<T> {
   }
 }
 
+/**
+ * Context methods are available when evaluating contextable SQL code, such as
+ * within SQLX files, or when using a [Contextable](#Contextable) argument with the JS API.
+ */
+export interface ITableContext extends ICommonContext {
+  /**
+   * Shorthand for an `if` condition. Equivalent to `cond ? trueCase : falseCase`.
+   * `falseCase` is optional, and defaults to an empty string.
+   */
+  when: (cond: boolean, trueCase: string, falseCase?: string) => string;
+
+  /**
+   * Indicates whether the config indicates the file is dealing with an incremental table.
+   */
+  incremental: () => boolean;
+}
+
 export class LegacyConfigConverter {
+  // This is a workaround to make bigquery options output empty fields with the same behaviour as
```

Comment:
do we have a test that covers this?

Reply:
Yep! Most table tests actually touch this, for example the fact there's no bigquery field here: https://github.com/dataform-co/dataform/blob/0e98564706536c940111f7420694cd8a9f784c33/core/main_test.ts#L1453